mlfinlab/feature_importance/fingerpint.py

```diff
@@ -0,0 +1,259 @@
+"""
+Implementation of an algorithm described in Yimou Li, David Turkington, Alireza Yazdani
+'Beyond the Black Box: An Intuitive Approach to Investment Prediction with Machine Learning'
+(https://jfds.pm-research.com/content/early/2019/12/11/jfds.2019.1.023)
+"""
+
+from abc import ABC, abstractmethod
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.linear_model import LinearRegression
+
+
+# pylint: disable=invalid-name
+
+class AbstractModelFingerprint(ABC):
```

Comment:
I was using the API yesterday and playing around with it. 

Lets follow the Sklearn API style where we we do the following:
```
fingerprint = RegressionFingerprint()
fingerprint.fit(est, X, num_values, pairwise_combinations=None)
linear, nonlinear, pairwise = fingerprint.get_effects()
```

Reply:
I really like the idea. The only problem which I see right now is that fit() code will be overcrowded with a code blocks from __init__ methods (because they need information from clf and X)

Reply:
The `init` method should be used to specify and set class attributes.

The chunk of code from `# Get possible feature values (Step 1)` can be split up into concise methods which can then be called from within the fit method. This should resolve the problem and improve readability?

Reply:
i only see libs using a extension of sklearn api when using predict() at fit i don't see a **kwarg param, this break the sklearn api in some way, but well... does we will use it in a gridsearch or pipeline?

Reply:
@rspadim, I am not sure I understand your question?